realm/realm-library/src/androidTest/java/io/realm/LinkingObjectsDynamicTests.java

```diff
@@ -20,34 +20,484 @@
+        final RealmResults<DynamicRealmObject> target2Sources = target2.linkingObjects(BacklinksSource.CLASS_NAME, BacklinksSource.FIELD_CHILD);
+        assertNotNull(target2Sources);
+        assertEquals(numSourceOfTarget2, target2Sources.size());
+        for (DynamicRealmObject target2Source : target2Sources) {
+            assertEquals(BacklinksSource.CLASS_NAME, target2Source.getType());
+            assertTrue(target2Source.getString(BacklinksSource.FIELD_NAME).endsWith("_target2"));
+            assertEquals(target2, target2Source.getObject(BacklinksSource.FIELD_CHILD));
+        }
+
+        final DynamicRealmObject target3 = dynamicRealm.where(BacklinksTarget.CLASS_NAME).equalTo(BacklinksTarget.FIELD_ID, 3).findFirst();
+        final RealmResults<DynamicRealmObject> target3Sources = target3.linkingObjects(BacklinksSource.CLASS_NAME, BacklinksSource.FIELD_CHILD);
+        assertNotNull(target3Sources);
+        assertTrue(target3Sources.isEmpty());
+    }
+
+    @Test
+    public void linkingObjects_OBJECT_notDefinedInModel() {
+        final int numOwnersOfCat1 = 3;
+        final int numOwnersOfCat2 = 2;
+        realm.executeTransaction(new Realm.Transaction() {
+            @Override
+            public void execute(Realm realm) {
+                final Cat cat1 = realm.createObject(Cat.class);
+                cat1.setName("cat1");
+
+                // create owners of cat1
+                for (int i = 0; i < numOwnersOfCat1; i++) {
+                    final Owner owner = realm.createObject(Owner.class);
+                    owner.setName("owner" + i + "_cat1");
+                    owner.setCat(cat1);
+                }
+
+                final Cat cat2 = realm.createObject(Cat.class);
+                cat2.setName("cat2");
+
+                // create owners of cat2
+                for (int i = 0; i < numOwnersOfCat2; i++) {
+                    final Owner owner = realm.createObject(Owner.class);
+                    owner.setName("owner" + i + "_cat2");
+                    owner.setCat(cat2);
+                }
+
+                // cat3 has no owner
+                final Cat cat3 = realm.createObject(Cat.class);
+                cat3.setName("cat3");
+            }
+        });
+
+        final DynamicRealmObject cat1 = dynamicRealm.where(Cat.CLASS_NAME).equalTo(Cat.FIELD_NAME, "cat1").findFirst();
+        final RealmResults<DynamicRealmObject> cat1Owners = cat1.linkingObjects(Owner.CLASS_NAME, Owner.FIELD_CAT);
+        assertNotNull(cat1Owners);
+        assertEquals(numOwnersOfCat1, cat1Owners.size());
+        for (DynamicRealmObject cat1Owner : cat1Owners) {
+            assertEquals(Owner.CLASS_NAME, cat1Owner.getType());
+            assertTrue(cat1Owner.getString(Owner.FIELD_NAME).endsWith("_cat1"));
+            assertEquals(cat1, cat1Owner.getObject(Owner.FIELD_CAT));
+        }
+
+        final DynamicRealmObject cat2 = dynamicRealm.where(Cat.CLASS_NAME).equalTo(Cat.FIELD_NAME, "cat2").findFirst();
```

Comment:
@nhachicha yes. We should check if the only owners who point the cat are returned.
Just checking the owners of `cat1`is enough?